src/Analyser/ExpressionResult.php

```diff
@@ -28,6 +28,7 @@ public function __construct(
 		private array $impurePoints,
 		?callable $truthyScopeCallback = null,
 		?callable $falseyScopeCallback = null,
+		private bool $isAlwaysTerminating = false,
```

Comment:
Doesn't have to be optional and doesn't have to be at the end. I'd prefer it to be after `hasYield`, for consistency with StatementResult.